OSDOCS-3889 add NLB to AWS networking config (#51460)
openshift/openshift-docs: modules/installation-configuration-parameters.adoc

|The configuration for the specific platform upon which to perform the installation: `alibabacloud`, `aws`, `baremetal`, `azure`, `ibmcloud`, `nutanix`, `openstack`, `ovirt`, `vsphere`, or `{}`. For additional information about `platform.<platform>` parameters, consult the table for your specific platform that follows.
|Object

ifdef::aws[]

> |`lbType`

Reviewer: ```suggestion
|`platform.aws.lbType`
```
Use this formatting to denote 
```
platform:
    aws:
      lbType: nlb
```

|Specify one or more repositories that may also contain the same images.
|Array of strings

ifdef::aws[]
|`platform.aws.lbType`

> |Required to set the NLB load balancer type. The default value is Classic LB. If you use NLB load balancing, the ingress operator creates the load balancer for the default ingress controller.

Reviewer 2: It can either be "NLB" or "Classic" cc @Miciah

Author: @miheer And will the `oc edit ingress.config/cluster -o yaml` command have to be run to save or update the cluster configuration file that the installer outputs?

> |Required to set a load balancer type other than the default Classic, such as NLB. The ingress operator creates the load balancer for the default ingress controller based on the configuration used here.

Reviewer 2: Required to set a load balancer type in AWS. Valid values are Classic or NLB.

Reviewer 2: For other ingress controllers than the default if you don't Specify Classic or NLB in the ingress controller CR then the lbType which was set during install will referred. The installer sets the lbType value in ingress cluster config object.

Author: @miheer A value in this installation parameter is only truly required to set the NLB load balancer type, is that correct? The customer has the option to leave this field blank during installation and the installer will set lbType to Classic by default--am I understanding correctly? (Per the .yaml example.) Edits coming....TY for reviewing.

> |Required to set the NLB load balancer type in AWS. Valid values are `Classic` or `NLB`. The default is Classic if no value is provided. The installation program sets the value provided here in the ingress cluster config object. If you do not specify a load balancer type for other ingress controllers, they will use the configuration set in this parameter.

Reviewer 3: ```suggestion
|Required to set the NLB load balancer type in AWS. Valid values are `Classic` or `NLB`. If no value is specified, the installation program defaults to `Classic`. The installation program sets the value provided here in the ingress cluster configuration object. If you do not specify a load balancer type for other ingress controllers, they use the type set in this parameter.
```

- See the comment above on API object formatting. 
- The ISG recommends [avoiding future tense](https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#tense).
- See the [ISG guidance](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#config) on config. _config map_ and _build config_ are listed as exceptions in the [SSG](https://redhat-documentation.github.io/supplementary-style-guide/).
- The suggestion to use _type_ instead of _configuration_ in the last sentence is optional. It just sounds a little better to my ear.

> |Letters `NLB`.

Reviewer 4: The values in the 3rd column of the table have to be expressed as values used by the installer program. There's no "Letters" value, it's going to be an ASCII "String" value. I recommend replacing the word "Letters" with "String" and include the word "Classic" also. I recommend the following: "Strings 'Classic' or 'NLB' "

> |String 'Classic' or 'NLB'.

Reviewer 3: ```suggestion
|`Classic` or `NLB`. The default value is `Classic`.
```
I think this is the design pattern used in the install docs. @bscott-rh Would you please confirm?

Also, backticks, not single quotes, for literals.
